iceoryx_posh/doc/chunk_header.md

```diff
@@ -0,0 +1,218 @@
+maxAlignment = max(alignof(payloadOffset), payloadAlignment);
+chunkSize = sizeLeftOfThePayloandAlignmentBoundary + maxAlignment + payloadSize;
+```
+
+### Accessing Chunk Header Extension
+
+The `ChunkHeader` has a template method to get custom header. There is a risk to use this wrong by accident, but there is currently no better solution for this.
+
+Since the custom header is always adjacent to the `ChunkHeader`, the formula to get the custom header is:
+```
+customHeader = addressOf(chunkHeader) + sizeof(chunkHeader);
+```
+
+### ChunkHeader Methods
+
+- `void* payload()` returns a pointer to the payload
+- `template <typename T> T* customHeader()` returns a pointer to the custom header
+- `static ChunkHeader* fromPayload(const void* const payload)` returns a pointer to the `ChunkHeader` associated to the payload
+
+### Integration Into Publisher/Subscriber API
+
+The `Publisher` has additional template parameters for the custom header and payload alignment. Alternatively this could also be done with the `allocate` method, but that increases the risk of using it wrong by accident.
+
+Additionally, to fully integrate the custom header into the regular control flow, a pointer to a `ChunkHeaderHook` can be passed to the `Publisher` c'tor.
+
+The `ChunkHeaderHook` is not in the shared memory and has the following virtual methods:
+- `virtual void allocateHook(ChunkHeader& chunkHeader)` can be used to initialize the custom header
+- `virtual void deliveryHook(ChunkHeader& chunkHeader)` can be used to set members of the custom header, e.g. a timestamp
+
+Furthermore, the `Publisher` and `Subscriber` have access to the `ChunkHeader` and can use the `customHeader()` method to gain access to the custom header.
+
+### Pitfalls & Testing
+
+- when the payload is adjacent to the `ChunkHeader`, it must be ensured that `m_payloadOffset` overlaps with the `back-offset`, which is `sizeof(m_payloadOffset)` in front of the payload
+- to simplify calculation, it is assumed that the alignment of the custom header doesn't exceed the alignment of the `ChunkHeader`. This has to be enforced with an `assert`
+
+# Open Issues
+
+- the design was done with the intention to have a custom header of arbitrary size, if the size is limited to e.g. 32 bytes, some things could be simplified
+- untyped API and C bindings
+    - PoC is needed
+- user defined sequence number
+    - this can probably be done by a `ChunkHeaderHook`
+    - alternatively, a flag could be introduce into the `ChunkHeader`
+- user defined timestamp
+    - this should probably be in a custom header
+- mempool configuration
+    - currently we specify the payload size and the size of the `ChunkHeader` is added automatically
+    - with the new approach, part of the specified payload size might be used for the custom header
+    - part of the specified payload might also be used as padding for the payload alignment
+    - shall we change this back to just specifying the full chunk size?
```

Comment:
I would keep it as simple as possible and would just specify the full chunk size.